torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -300,8 +300,7 @@ class ProcessGroupNCCL : public ProcessGroup {
 
       (*cudaEvents_)[0].block(*futureNCCLCallbackStream_);
       // Use the dedicated callback stream to run callback.
-      c10::OptionalStreamGuard streamGuard{
-          c10::Stream(*futureNCCLCallbackStream_)};
+      c10::StreamGuard streamGuard{*futureNCCLCallbackStream_};
```

Comment:
I also changed this so that there's a formal guarantee that we'll set a stream (i.e., remove the "Optional" part)